broker/data-access-layer/eventmesh/src/ApiServerClient.js

```diff
@@ -77,24 +75,49 @@ function convertToHttpErrorAndThrow(err) {
 }
 
 class ApiServerClient {
-  constructor() {	
+  constructor() {
     this.ready = false;
+    this.apiClients = {};
     this.init();
   }
 
   init() {
-    return Promise.try(() => {
-      return Promise.map(_.values(config.apiserver.crds), crdTemplate => {
-        apiserver.addCustomResourceDefinition(yaml.safeLoad(Buffer.from(crdTemplate, 'base64')));
-      })
-        .tap(() => {
-          logger.debug('Successfully added enpoints to apiserver client');
-        })
-        .catch(err => {
-          logger.error('Error occured while adding enpoints to apiserver client', err);
-          return convertToHttpErrorAndThrow(err);
-        });
-    });
+    this.apiserverConfig = new k8s.KubeConfig();
+    if (config.apiserver.getConfigInCluster) {
+      this.apiserverConfig.loadFromCluster();
+    } else if (config.apiserver.pathToKubeConfig) {
+      this.apiserverConfig.loadFromFile(config.apiserver.pathToKubeConfig);
+    } else {
+      assert.fail('Config \'apiserver.pathToKubeConfig\' must be provided if \'apiserver.getConfigInCluster\' is false');
+    }
+    this.watch = new k8s.Watch(this.apiserverConfig);
+  }
+
+  _getApiClient(resourceGroup, version) {
+    let apiClientType;
+    let apiType;
+    version = _.join(_.split(_.startCase(version), ' '), '');
+
+    // Use CustomObjectsApi for all apis except core api
+    switch (resourceGroup) {
+      case '':
+        apiType = 'Core' + version + 'Api';
```

Comment:
Can we make them constants ? and please see if they are already defined and reuse in that case

Reply:
These are class names of various clients in k8s sdk. I used
```
apiType = 'Apiextensions' + version + 'Api';
apiClientType = k8s[apiType];
```
instead of just
```
apiClientType = k8s.ApiextensionsV1Api
```
thinking if we somhow need the old `v1beta1` api also. 

I will remove this version manipulation and make the mapping from `resrourceGroup` to client name as constant